legal-api/src/legal_api/services/filings/validations/alteration.py

```diff
@@ -0,0 +1,58 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Alteration filing."""
+from http import HTTPStatus
+from typing import Dict
+
+from flask_babel import _ as babel
+
+from legal_api.errors import Error
+from legal_api.models import Business, Filing
+
+from ... import namex
+from ...utils import get_str
+
+
+def validate(business: Business, filing: Dict) -> Error:
+    """Validate the Alteration filing."""
+    if not business or not filing:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': _('A valid business and filing are required.')}])
+    msg = []
+
+    nr_path = '/filing/alteration/nameRequest/nrNumber'
+    nr_number = get_str(filing.json, nr_path)
+
+    # ensure NR is approved or conditionally approved
+    nr_response = namex.query_nr_number(nr_number)
```

Comment:
Is it possible alter a company without a NR? Ie, business will get "BC xxx" number instead? (Please confirm with BA.)

If so then this code needs to allow it.

Reply:
In Name Change for Alteration it doesn't have a NR when the change is a numbered company. e.g. 12345678 B.C. Ltd.

Reply:
The ticket said:

> ensure only "change of Name" NR request types can be used for corrections and alterations

Who wold I double check that with?

Reply:
You can double-check with the ticket author (Sienna).

Reply:
I asked Linda and Sienna and they told me that there's no name request when changing to a numbered company.

Reply:
I'll update the validator to accommodate that then